lib/matplotlib/figure.py

```diff
@@ -336,6 +336,7 @@ def __init__(self,
             raise ValueError('figure size must be finite not '
                              '{}'.format(figsize))
         self.bbox_inches = Bbox.from_bounds(0, 0, *figsize)
+        self.figsize = figsize
```

Comment:
Why do we need these lines?  How did these used to be handled?

Reply:
Because these parameters have to be accessed by clf, outside the scope of __init__. Before, they were only used within that scope.